### PR TITLE
added test matrix

### PR DESCRIPTION
**Pull request by billingsley-john**
## Proposed changes

PR to add matrix testing to github actions CI for different cadquery versions 

## Types of changes

What types of changes does your code introduce to the Paramak?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Code refactoring
- [ ] Documentation Update (if none of the other choices apply)
- [x] New tests

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [ ] Pep8 applied
- [ ] Unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by shimwell** [comment]
That was quick

File: .github/workflows/docker_ci.yml [hunk 1]
```diff
@@ -8,10 +8,19 @@ on:
 jobs:  
   build:     
    runs-on: ubuntu-latest
+   strategy:
+    matrix:
+      include:
+        - cadquery-version: 2.1
+        - cadquery-version: master
```
Comment (shimwell) [inline .github/workflows/docker_ci.yml]:
```suggestion
        - cadquery-version: [2.1, master]
```

**Comment by shimwell** [comment]
Looks like the simulations using stl files made in cadquery version 2.1 work and the  stl files made in cadquery master version doesn't work.

**Comment by billingsley-john** [comment]
> Looks like the simulations using stl files made in cadquery version 2.1 work and the stl files made in cadquery master version doesn't work.

I wonder at what point this stopped working. I suppose this managed to get through because the circle ci tests are pulling a static docker image - we are already seeing the benefits of this new testing approach.

**Comment by billingsley-john** [comment]
Currently, when one set of tests in the matrix fails, it causes the cancellation of the other tests in the matrix. I.e. if tests fail for master, the tests running for 2.1 are cancelled. We don't want this - we want to return pass/fail for both scenarios each time.

**Comment by billingsley-john** [comment]
Continued in #764

**Pull request closed without merging**